python/pyarrow/src/arrow/python/inference.cc

```diff
@@ -27,7 +27,9 @@
 #include <utility>
 #include <vector>
 
+#include "arrow/scalar.h"
 #include "arrow/status.h"
+#include "arrow/type.h"
```

Comment:
Is this additional include needed? (we already use shared pointers to DataType in this file, so I would expect your diff to not need this)

Reply:
Correct, I guess it must be a leftover. Will remove it 👍